src/TriggerActivityFactory.cpp

```diff
@@ -0,0 +1,43 @@
+/* @file: TriggerActivityFactory.cpp
+ *
+ * This is part of the DUNE DAQ Application Framework, copyright 2023.
+ * Licensing/copyright details are in the COPYING file that you should have
+ * received with this code.
+ */
+#include <iostream>
+
+#include "triggeralgs/TriggerActivityFactory.hpp"
+#include "TRACE/trace.h"
+#define TRACE_NAME "TriggerActivityFactory"
+
+using namespace triggeralgs;
+
+TriggerActivityFactory::TAMakerMap& TriggerActivityFactory::getTAMakers() {
+  static TAMakerMap s_makers;
+  return s_makers;
+}
+
+bool TriggerActivityFactory::registerCreator(const std::string algName, TAMakerCreator creator)
+{
+  TAMakerMap& makers = getTAMakers();
+
+  auto it = makers.find(algName);
+  if (it == makers.end()) {
+    makers[algName] = creator;
+    return true;
+  }
+
```

Comment:
Maybe some error message? If we're trying to overwrite an already-registered algorithm, then we've done something horribly wrong.